packages/cli/src/util/client.ts

```diff
@@ -140,19 +135,31 @@ export default class Client extends EventEmitter {
         return res;
       }
 
-      if (!res.headers.get('content-type')) {
+      const contentType = res.headers.get('content-type');
+      if (!contentType) {
         return null;
       }
 
-      return res.headers.get('content-type').includes('application/json')
-        ? res.json()
-        : res;
+      return contentType.includes('application/json') ? res.json() : res;
     }, opts.retry);
   }
 
-  _onRetry(error: Error) {
-    this.output.debug(`Retrying: ${error}\n${error.stack}`);
-  }
+  reauthenticate = sharedPromise(async function (
```

Comment:
This sharedPromise() is an interesting solution. I wonder if it can be solved with `async-sema` of 1 since we already have that

Reply:
Perhaps. I kinda like the current approach though since it doesn't need any dependencies.